core/src/stores/postgres.rs

```diff
@@ -3023,6 +3023,138 @@ impl Store for PostgresStore {
+        let mut params: Vec<&(dyn ToSql + Sync)> = vec![];
+
+        where_clauses.push("data_source = $1".to_string());
+        params.push(&data_source_row_id);
+
+        // Don't support tags for folders yet.
+        let view_filter = match view_filter {
+            None => None,
+            Some(filter) => Some(SearchFilter {
+                tags: None,
+                parents: filter.parents.clone(),
+                timestamp: filter.timestamp.clone(),
+            }),
+        };
+
+        let (filter_clauses, filter_params, mut p_idx) =
+            Self::where_clauses_and_params_for_filter(&view_filter, params.len() + 1);
+
+        where_clauses.extend(filter_clauses);
+        params.extend(filter_params);
+
+        // Add table_ids filter if provided.
+        if let Some(ref ids) = folder_ids {
+            // Create a dynamic list of placeholders for the table IDs.
+            let id_placeholders: Vec<String> = (0..ids.len())
+                .map(|_| {
+                    let placeholder = format!("${}", p_idx);
+                    p_idx += 1; // Increment p_idx after each table.
+                    placeholder
+                })
+                .collect();
+
+            where_clauses.push(format!("node_id IN ({})", id_placeholders.join(", ")));
+            params.extend(ids.iter().map(|id| id as &(dyn ToSql + Sync)));
+        }
+
+        let sql = format!(
+            "SELECT node_id, title, timestamp, parents \
+               FROM data_sources_nodes \
+               WHERE folder IS NOT NULL AND {} ORDER BY timestamp DESC",
+            where_clauses.join(" AND "),
+        );
+
+        let rows = match limit_offset {
+            None => {
+                let stmt = c.prepare(&sql).await?;
+                c.query(&stmt, &params).await?
+            }
+            Some((limit, offset)) => {
+                let limit = limit as i64;
+                let offset = offset as i64;
+
+                let mut params = params.clone();
+                params.push(&limit);
+                params.push(&offset);
+
+                let stmt = c
+                    .prepare(&(sql + &format!(" LIMIT ${} OFFSET ${}", p_idx, p_idx + 1)))
+                    .await?;
+                c.query(&stmt, &params).await?
+            }
+        };
+
+        let folders: Vec<Folder> = rows
+            .into_iter()
+            .map(|r| {
+                let node_id: String = r.get(0);
+                let title: String = r.get(1);
+                let timestamp: i64 = r.get(2);
+                let parents: Vec<String> = r.get(3);
+
+                Ok(Folder::new(
+                    project,
+                    &data_source_id,
+                    &node_id,
+                    timestamp as u64,
+                    &title,
+                    parents,
+                ))
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        let total = match limit_offset {
```

Comment:
don't understand why we do a count by sql here (and nit: is it possible to merge it with the match clause above)?